src/x_misc.c

```diff
@@ -56,7 +58,7 @@ typedef struct _random
 
 static int makeseed(void)
 {
-    static unsigned int random_nextseed = 1489853723;
+    unsigned int random_nextseed = time(NULL) * ++instanceindex * 151;
     random_nextseed = random_nextseed * 435898247 + 938284287;
```

Comment:
The original implementation of `makeseed` is actually a PRNG by itself. The only thing we have to do is replace the hardcoded initial value with a call to `time(NULL)`. 
```
static int makeseed(void)
{
        /* this is not thread-safe, but it does not matter for our purposes */
    static unsigned int random_nextseed = 0;
    if (!random_nextseed)
        random_nextseed = time(NULL);
    random_nextseed = random_nextseed * 435898247 + 938284287;
    return (random_nextseed & 0x7fffffff);
}
```

---

Sidenote: in C++11 you could simply write the following and it would even be thread-safe:
```
static unsigned int random_nextseed = time(NULL);
```

Reply:
Also, we only have to implement `makeseed` in `x_random` and make it non-static, so we can reuse it for `[array random]` and `[noise~]`.

Reply:
> BTW, it does not really make sense to implement the same function three times. Instead, makeseed() 

it's not exactly the same for [noise~] but it's the same for [random] and [array random]

Reply:
> The only thing we have to do is replace the hardcoded initial value with a call to time(NULL).

this wouldn't work to create different seed values for each instance and wouldn't work for a 'seed' message without an argument to reset the seed according to current time.

Reply:
>  Instead, makeseed() in x_random.c could be non-static 

the instanceindex variable would also need to be non-static, right? But anyway, before things were also exactly the same for different objects and it was fine, it just wasn't in a separate function. Actually, it was a separate function in [random] but it was used just once, I then made it be used twice, which makes more sense I think.

Reply:
> it's not exactly the same for [noise~] but it's the same for [random] and [array random]

There is no reason why `[noise~]` would need a different seed method.

Reply:
> this wouldn't work to create different seed values for each instance

it does work because `makeseed` is really a PRNG on its own and every call yields a different value. the only difference is that we *initialize* the sequence with a pseudo-random number (= `time()`) instead of using a hardcoded value.

>  wouldn't work for a 'seed' message without an argument to reset the seed according to current time.

it does work because `makeseed` returns an unpredictable number.

> the instanceindex variable would also need to be non-static, right?

in my function there is no instance index variable ;-)

Reply:
BTW, I just edited my implementation of `makeseed`: `random_nextseed ` is now `PERTHREAD` to make it reentrant.

Reply:
So, copy your makeseed "static" code into radom's code, call it in [noise~] and [array random], as simple as that, right? Will try it!

Reply:
btw, what is pd's method for generating pseudo random numbers? Doesn't seem to be "Linear Congruential Generator", but it's similar and maybe that's it?

Reply:
I'm not succeeding in making the bridge and telling where to look for "makeseed".

I get errors:

Undefined symbols for architecture x86_64:
  "_makeseed", referenced from:
      _noise_new in pd-d_osc.o
      _noise_seed in pd-d_osc.o
      _arrayobj_new in pd-x_array.o
      _array_random_new in pd-x_array.o
      _array_random_seed in pd-x_array.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

I tried declaring "static int makeseed();" or just "int makeseed();" in the code of noise~/array...

Reply:
hmm, maybe I got it now